WordPressKit/StatsServiceRemoteV2.swift

```diff
@@ -0,0 +1,193 @@
+
+        let path = self.path(forEndpoint: "sites/\(siteID)/stats/post/\(postID)", withVersion: ._1_1)
+
+        wordPressComRestApi.GET(path,
+                                parameters: parameters,
+                                success: { (response, _) in
+                                    guard
+                                        let jsonResponse = response as? [String: AnyObject],
+                                        let views = jsonResponse["views"] as? Int else {
+                                            completion(nil, ResponseError.decodingFailure)
+                                            return
+                                    }
+                                    completion(views, nil)
+                                }, failure:  { (error, _) in
+                                    completion(nil, error)
+                                }
+        )
+    }
+
+}
+
+// This serves both as a way to get the query properties in a "nice" way,
+// but also as a way to narrow down the generic type in `getInisght:` method.
+public protocol InsightProtocol {
+    static var queryProperties: [String: AnyObject] { get }
+    static var pathComponent: String { get }
+
+    init?(jsonDictionary: [String: AnyObject])
+}
+
+extension InsightProtocol {
+
+    // A big chunk of those use the same endpoint and queryProperties.. Let's simplify the protocol conformance in those cases.
+
+    public static var queryProperties: [String: AnyObject] {
+        return [:]
+    }
+
+    public static var pathComponent: String {
+        return "stats/"
+    }
+}
+
+// For some god-forsaken reason Swift compiler freaks out if this is not declared _in this file_,
+// and refuses to compile the project.
+// I'm guessing this has somethign to do with generic specialisation, but I'm not enough
+// of a `swiftc` guru to really know. Leaving this in here to appease Swift gods.
+// TODO: see if this is still a problem in Swift 5 mode!
+public struct StatsLastPostInsight {
+    public let title: String
+    public let url: URL
+    public let publishedDate: Date
+    public let likesCount: Int
+    public let commentsCount: Int
+    public let viewsCount: Int
+    public let postID: Int
+}
+
+extension StatsLastPostInsight: InsightProtocol {
+
+    //MARK: - InsightProtocol Conformance
+    public static var queryProperties: [String: AnyObject] {
+        return ["order_by": "date" as AnyObject,
+                "number": "1" as AnyObject,
+                "type": "post" as AnyObject,
+                "fields": "ID, title, URL, discussion, like_count, date" as AnyObject]
+    }
+
+    public static var pathComponent: String {
+        return "posts/"
+    }
+
+    public init?(jsonDictionary: [String: AnyObject]) {
+        fatalError("This shouldn't be ever called, instead init?(jsonDictionary:_ views:_) be called instead.")
+    }
+
+    //MARK: -
+
+    private static let dateFormatter: ISO8601DateFormatter = .init()
```

Comment:
❤️ I have yet to use this since it was added to iOS.

Reply:
It's a life-saver.